### PR TITLE
feat: Forbid task metadata and add upload functions

### PR DESCRIPTION
**Pull request by Samoed**
## Checklist

- [X] Run tests locally to make sure nothing is broken using `make test`. 
- [X] Run the formatter to format the code using `make lint`. 

Some retrieval tasks need to be reuploaded because they are loaded from different repositories. I’ve created upload functions to convert these datasets into our current format. I tested each reuploaded dataset, and the scores matched, except for `mFollowIR` (`rus`). In the main branch, the `main_score` is `-0.039465099069488106`, whereas the reuploaded dataset gives `-0.031187925634321677`. However, this run was only for testing purposes.

Initially, I tried adding this script to the `mteb` folder, but it gave an error: `AttributeError: module 'logging' has no attribute 'getLogger'`. So, I moved it to the `scripts` folder.

Additionally, some tasks may not be imported successfully. For example, I tried to load `IndicXnliPairClassification`, but it resulted in an error.

[uploaded.zip](https://github.com/user-attachments/files/17968625/uploaded.zip)
[mteb main.zip](https://github.com/user-attachments/files/17968622/mteb.main.zip)

### CODE REVIEW

**Comment by KennethEnevoldsen** [comment]
@Samoed is this PR stale?

**Comment by Samoed** [comment]
Yes, some datasets (mostly from CMTEB) load data from multiple repositories on HF, so we need to convert them first to complete this PR.

**Comment by Samoed** [comment]
Will try to add new

**Comment by orionw** [comment]
I’m traveling and won’t be at a computer til the end of the week, but this looks good.

Are there any datasets that are still not converted?

And is the mFollowIR Russian still an issue? FWIW the v2 branch fixed a small bug that the current one doesn’t have, so the numbers from main and v2 will be different.  The number looks reasonable and I wouldn’t worry about it.

**Comment by Samoed** [comment]
I've tested from `v2.0.0` branch, but mostly this is not the issue, because I just tested how uploader would work with multilingual tasks. I didn't change `mFollowIR` dataset version

**Comment by Samoed** [comment]
@KennethEnevoldsen Can you review, please?

**Review by KennethEnevoldsen — COMMENTED** [review]
A few minor things and a suggestion to move the upload utility to the class object (assuming we want to maintain it)

Generally though this looks great!

File: tests/test_TaskMetadata.py [hunk 1]
```diff
@@ -511,10 +499,11 @@ def test_disallow_trust_remote_code_in_new_datasets():
         "MLSUMClusteringS2S.v2",
         "SwednClusteringP2P",
         "SwednClusteringS2S",
+        "IndicXnliPairClassification",
```
[the remark this answers is not in the document]
Comment (Samoed) [reply tests/test_TaskMetadata.py:502]: Previously this wasn't added to `__init__.py` file

[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply tests/test_TaskMetadata.py:502]: ahh right - perfect

**Review by KennethEnevoldsen — APPROVED** [review]
Loving it <3!

**Pull request merged**